detox/src/ipc/IPCClient.js

```diff
@@ -109,8 +109,8 @@ class IPCClient {
     });
   }
 
-  _onSessionStateUpdate = (payload) => {
-    this._state.patch(payload);
+  _onSessionStateUpdate = ({ testResults, workersCount }) => {
```

Comment:
Wait, no. It was meant to be generic.
Can we return that back, please?

Reply:
I can. why do you want that to be generic if we're using only the `testResults` and the `workersCount`?

Reply:
I think it's better to limit only for the props that you want to be able to change from this place.

Reply:
It is going to turn into a source of errors. The contract is that the IPC server itself knows what properties it should update/broadcast to all the clients. The client should be obedient in that regard.

Reply:
How the state's props `id`, `detoxConfig` or `detoxIPCServer` are relevant here? I really see no reason to allow their change from here.

Reply:
> It is going to turn into a source of errors. The contract is that the IPC server itself knows what properties it should update/broadcast to all the clients. The client should be obedient in that regard.

OK, I get that.

Reply:
Yes, but I see no reason to update them from the server. It does not send the entire session object – only the changed parts of it.

Reply:
I can understand what you're saying from this perspective, but I still think the API from both the client and the server should be restricted only for what the server can or should update. 

I'll do the revert because I don't think this discussion worth the bikeshedding 😄

Reply:
done